apps/gallery/test/marionette/activity_test.js

```diff
@@ -33,128 +33,165 @@ marionette('Gallery Activity Tests', function() {
+        return activityTesterApp.sharedImageName.displayed();
+      });
 
-    client.waitFor(function(){
-      return activityTesterApp.sharedImageName.displayed();
-    });
+      // Compare shared blob name, type and size with the
+      // image loaded in gallery app
+      var sharedImageName = activityTesterApp.sharedImageName.text();
+      assert.strictEqual(sharedImageName, imageInfo.name);
 
-    // Compare shared blob name, type and size with the
-    // image loaded in gallery app
-    var sharedImageName = activityTesterApp.sharedImageName.text();
-    assert.strictEqual(sharedImageName, imageInfo.name);
+      var sharedImageType = activityTesterApp.sharedImageType.text();
+      assert.strictEqual(sharedImageType, imageInfo.type);
 
-    var sharedImageType = activityTesterApp.sharedImageType.text();
-    assert.strictEqual(sharedImageType, imageInfo.type);
+      var sharedImageSize = activityTesterApp.sharedImageSize.text();
+      assert.strictEqual(sharedImageSize, imageInfo.size);
+    });
 
-    var sharedImageSize = activityTesterApp.sharedImageSize.text();
-    assert.strictEqual(sharedImageSize, imageInfo.size);
-  });
+    test('open memory backed blob using gallery open activity', function() {
+      activityTesterApp.launch();
+      // Initiate open activity by clicking open image button.
+      // This will generate memory backed blob using canvas
+      // and trigger open activity to view the blob
+      activityTesterApp.tapOpenImageButton();
 
+      system.switchToApp(Gallery.ORIGIN);
 
-  test('open memory backed blob using gallery open activity', function() {
-    activityTesterApp.launch();
-    // Initiate open activity by clicking open image button.
-    // This will generate memory backed blob using canvas
-    // and trigger open activity to view the blob
-    activityTesterApp.tapOpenImageButton();
+      // Check there is no filename in title bar
+      var title = fullscreenView.openActivityImageTitle.text();
+      assert.strictEqual(title, '');
 
-    system.switchToApp(Gallery.ORIGIN);
+      // Check save button is not displayed
+      assert.ok(!fullscreenView.openActivitySaveButton.displayed(),
+        'Save button is not displayed');
 
-    // Check there is no filename in title bar
-    var title = fullscreenView.openActivityImageTitle.text();
-    assert.strictEqual(title, '');
+      client.waitFor(function(){
+        return fullscreenView.openActivityImage.displayed();
+      });
 
-    // Check save button is not displayed
-    assert.ok(!fullscreenView.openActivitySaveButton.displayed(),
-      'Save button is not displayed');
+      // Check if displayed image src is set with blob url
+      assert.ok(fullscreenView.hasSrcImageBlobURL(Gallery.ORIGIN,
+        fullscreenView.openActivityImage));
+     });
+  });
 
-    client.waitFor(function(){
-      return fullscreenView.openActivityImage.displayed();
+  suite('Gallery app multiple images', function() {
```

Comment:
In this case, since you're not deleting images, I'm not sure that it is necessary to have two different suites for the 1 photo and 2 photo cases.  What if you just modified your existing suite to have two photos, and then put this one new test into that existing suite.  Would that break any other tests?  It would certainly make the diff smaller.